engine/websocketroutine_manager.go

```diff
@@ -287,6 +284,9 @@ func (m *WebsocketRoutineManager) websocketDataHandler(exchName string, data int
 		}
 		m.syncer.PrintOrderbookSummary(base, "websocket", nil)
 	case *order.Detail:
+		if !m.orderManager.IsRunning() {
+			return nil
+		}
 		if !m.orderManager.Exists(d) {
 			err := m.orderManager.Add(d)
 			if err != nil {
```

Comment:
While I understand what you're saying regarding the `nil` and the interface side of things, I'm unable to actually get a panic. Are you able to expand on that?
I'm running `--ordermanager=false`, it is not setup, and it is able to correctly fail `!m.orderManager.Exists(d)` on account of `m == nil`. Then after `err := m.orderManager.Add(d)` I receive the error 
<img width="692" alt="image" src="https://github.com/thrasher-corp/gocryptotrader/assets/9261323/f6e07e83-8309-48d9-aafc-397eff131b26">
which is only hit from
```go
func (m *OrderManager) Add(o *order.Detail) error {
	if m == nil {
		return fmt.Errorf("order manager %w", ErrNilSubsystem)
	}
```

I'm happy with the changes, but I want to ensure there is no panic remaining, but don't experience a panic on `master` myself

Reply:
I'll have to hunt back; this work was actually a few months ago originally.
It's not impossible I'm wrong it was panicing, and it was just throwing this error. Let me check.

Reply:
Yep; I can't figure it out either. I had a note that it panics.
All public methods inside it protect against nil. I can't see why or how it'd panic.

Ironically I broke one of my commit message rules.
Instead of the message saying what the bug would look like and how to reproduce, I focused on the intricasies of nil interface arguments.

As you say, though, this is valid either way, even if it doesn't panic.